packages/babel-helpers/src/helpers.js

```diff
@@ -588,28 +588,46 @@ helpers.interopRequireDefault = helper("7.0.0-beta.0")`
 `;
 
 helpers.interopRequireWildcard = helper("7.0.0-beta.0")`
+  function _getRequireWildcardCache(obj) {
+    if (typeof global === 'undefined' || typeof WeakMap !== 'function') {
+      return null;
+    }
+
+    if (!global._REQUIRE_WILD_CACHE) {
+      global._REQUIRE_WILD_CACHE = new WeakMap();
+    }
+
+    return global._REQUIRE_WILD_CACHE;
+  }
```

Comment:
We can make it work when `global` is not defined but `@babel/runtime` is used (e.g. in rollup or webpack).
Also, could you rename the global var so that it is clear that comes from Babel?

```js
function _getRequireWildcardCache() {
  if (typeof WeakMap !== "function") return null;

  var cacheKey = "##__BABEL__REQUIRE__WILDCARD_CACHE";
  var map;
  if (typeof global !== "undefined") {
    if (global[cacheKey]) map = global[cacheKey];
    else map = global[cacheKey] = new WeakMap();
  } else {
    map = new WeakMap();
  }

  _getRequireWildcardCache = function() { return map };
  return map;
}
```

By doing so, multiple calls to `_getRequireWildcardCache` will return the same WeakMap even if `global` is not available.

Reply:
@nicolo-ribaudo Your approach looks pretty cool. 

Come to think of it again, If we use function reassigning, there seems to be no reason to use `global`. I changed the implementation. what do you think about it?

Reply:
The global map would still be needed if someone has two different instances of this helper.
I prefer this as you currently implemented it (since it doesn't rely on `global`), even if it doesn't optimize 100% of the cases.